### ISSUE TITLE
Calculation of train and val indices in mean_std_dataset.py

### ISSUE THREAD

**Issue by calcasey** [JacobHanimann/scDINO#4]
Hi, thank you so much for writing scDINO! 

I am running the full pipeline including calculating mean and std of my training dataset before training the model. I think the train and val indices might be defined the wrong way around in mean_std_dataset.py. Currently the train and val indices are defined as `train_indices, val_indices = indices[split:], indices[:split]` where `split = int(numpy.floor(validation_split * dataset_size))` and `validation_split = 1-float(snakemake.params["fraction_for_mean_std"])`. The default value of `snakemake.params["fraction_for_mean_std"]` is currently 0.2.

This means that `split` represents 80 % of my full dataset size, so `train_indices` corresponds to the final 20 % of indices. With a dataset of 1200000 images, I get this output:

`length of dataset:  1200000`
`240000 960000`
`Train dataset consists of 240000 images.`

which I think is the wrong way around?

**Comment by JacobHanimann** [comment]
Hello there! I believe I did this intentionally. With snakemake.params["fraction_for_mean_std"], you specify the fraction of the dataset to be used for calculating the mean and standard deviation, rather than the entire training of the ViTs (this is set in the config file under meta: train_datasetsplit_fraction). So, if you set it to 0.2, the train_indices will correspond to 20% of the dataset for mean_std_dataset.py. I hope I'm correct, though I'm not entirely certain.

**Comment by calcasey** [comment]
Ooh, I see, so you use 20 % of the dataset to calculate the mean and standard deviation. Thanks for explaining! 

Should the 20 % of samples used to calculate mean and std be determined using a different seed than when I split the data into train/val images later in the pipeline, so that the mean and std is calculated from a mix of train/val images?

**Comment by JacobHanimann** [comment]
Sure sounds like a good idea
